### ISSUE TITLE
Poetry does not ignore Homebrew python

### ISSUE THREAD

**Issue by ifsheldon** [python-poetry/poetry#9540]
### Description

I'm transitioning from conda to poetry to manage my python dependencies and vitual envs. When I started a new project which needs python 3.11, it fails with the following

```shell
> poetry install
The currently activated Python version 3.12.4 is not supported by the project (~3.11).
Trying to find and use a compatible version.

Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command.
```

I've read the documentation, it should create a venv for me, _unless_ it detects one that is currently in use. So I ran 
```
> poetry env info

Virtualenv
Python:         3.12.4
Implementation: CPython
Path:           NA
Executable:     NA

Base
Platform:   darwin
OS:         posix
Python:     3.12.4
Path:       /opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12
Executable: /opt/homebrew/opt/python@3.12/Frameworks/Python.framework/Versions/3.12/bin/python3.12
```

This python is a dependency of one binary that was installed from homebrew. So, I wouldn't use it nor installing any dependencies with this Python. I can't uninstall it because I need the binary that depends on it. 

Poetry should just ignore the python installed as a dependency. 

### Workarounds

Or, at least Poetry should have to a way for us to tell it to ignore it. 

### Poetry Installation Method

pipx

### Operating System

macOS Sonoma

### Poetry Version

1.8.3

### Poetry Configuration

```bash session
cache-dir = "/Users/zhiqiu/Library/Caches/pypoetry"
experimental.system-git-client = false
installer.max-workers = null
installer.modern-installation = true
installer.no-binary = null
installer.parallel = true
keyring.enabled = true
solver.lazy-wheel = true
virtualenvs.create = true
virtualenvs.in-project = true
virtualenvs.options.always-copy = false
virtualenvs.options.no-pip = false
virtualenvs.options.no-setuptools = false
virtualenvs.options.system-site-packages = false
virtualenvs.path = "{cache-dir}/virtualenvs"  # /Users/zhiqiu/Library/Caches/pypoetry/virtualenvs
virtualenvs.prefer-active-python = false
virtualenvs.prompt = "{project_name}-py{python_version}"
warnings.export = true
```

### Example pyproject.toml

```TOML
[tool.poetry]
name = "graphrag"
version = "0.1.0"
description = ""
authors = ["maple <ifsheldon@outlook.com>"]
license = "MIT"
readme = "README.md"

[tool.poetry.dependencies]
python = "~3.11"
graphrag = "^0.1.1"


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

### Poetry Runtime Logs

```bash session
> poetry --vvv install

The currently activated Python version 3.12.4 is not supported by the project (~3.11).
Trying to find and use a compatible version.

Poetry was unable to find a compatible version. If you have one, you can explicitly use it via the "env use" command.
```

**Comment by dimbleby** [comment]
How would "ignoring" this python be different from what poetry is doing?

> Trying to find and use a compatible version.
> Poetry was unable to find a compatible version

The problem is not the 3.12 python, the problem is that poetry cannot find a 3.11 python

**Comment by ifsheldon** [comment]
@dimbleby 
OK, I think the understanding of venv is a bit different. 

My expectation is:
1. poetry finds homebrew's python and then ignore it
2. poetry creates a new venv that includes a new python 3.11 as I specified, so it should start downloading python 3.11 and then setup an environment for me.

A venv having its own Python feels natural to me, as I got used to conda.

What poetry is doing (as far as I understand) is:
a.  poetry finds homebrew's python, but the version mismatches
b. it needs to use a python 3.11 to create a venv, so it tries to find one but fails.

I was thinking poetry failed at step 1, but it actually failed at step b.

If my understanding is correct, then the problem then becomes: why can't poetry help me manage different versions of Python as well?
